File: src/test/java/com/baulsupp/oksocial/services/google/DiscoveryDocumentTest.java
package com.baulsupp.oksocial.services.google;

import com.google.common.io.Resources;
import java.io.IOException;
import java.net.URL;
import java.nio.charset.StandardCharsets;
import java.util.List;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static com.google.common.collect.Lists.newArrayList;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class DiscoveryDocumentTest {
  private DiscoveryDocument doc;

  @BeforeEach
  public void loadStaticIndex() throws IOException {
    URL url = DiscoveryDocumentTest.class.getResource("urlshortener.json");

    String definition = Resources.toString(url, StandardCharsets.UTF_8);

    doc = DiscoveryDocument.parse(definition);
  }

  @Test
  public void getUrlsFromFile() throws IOException {
    assertEquals("https://www.googleapis.com/urlshortener/v1/", doc.getBaseUrl());

    assertEquals(newArrayList("https://www.googleapis.com/urlshortener/v1/url",
        "https://www.googleapis.com/urlshortener/v1/url/history"), doc.getUrls());
  }

  @Test
  public void loadGmail() throws IOException {
    URL url = DiscoveryDocumentTest.class.getResource("gmail.json");

    String definition = Resources.toString(url, StandardCharsets.UTF_8);

    DiscoveryDocument gmailDoc = DiscoveryDocument.parse(definition);

    List<DiscoveryEndpoint> endpoints = gmailDoc.getEndpoints();

    for (DiscoveryEndpoint s : endpoints) {
      System.out.println(s.url());
    }
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/services/google/DiscoveryDocumentTest.kt
package com.baulsupp.oksocial.services.google

import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class DiscoveryDocumentTest {
  private var doc: DiscoveryDocument? = null

  @BeforeEach

  fun loadStaticIndex() {
    doc = DiscoveryDocument.parse(DiscoveryDocumentTest::class.java.getResource("urlshortener.json").readText())
  }

  @Test

  fun getUrlsFromFile() {
    assertEquals("https://www.googleapis.com/urlshortener/v1/", doc!!.baseUrl)

    assertEquals(listOf("https://www.googleapis.com/urlshortener/v1/url",
        "https://www.googleapis.com/urlshortener/v1/url/history"), doc!!.urls)
  }

  @Test

  fun loadGmail() {
    val gmailDoc = DiscoveryDocument.parse(DiscoveryDocumentTest::class.java.getResource("gmail.json").readText())

    val endpoints = gmailDoc.endpoints

    for (s in endpoints) {
      println(s.url())
    }
  }
}
